cirq-core/cirq/ops/clifford_gate.py

```diff
@@ -158,121 +158,131 @@ class CommonCliffordGateMetaClass(value.ABCMetaImplementAnyOneOf):
+                from_xz(x_to=pZ, z_to=pX),  # Z+X aka H
+                from_xz(x_to=pY, z_to=mZ),  # X+Y
+                from_xz(x_to=mX, z_to=pY),  # Y+Z
+                from_xz(x_to=mZ, z_to=mX),  # Z-X
+                from_xz(x_to=mY, z_to=mZ),  # X-Y
+                from_xz(x_to=mX, z_to=mY),  # Y-Z
+                # 16..23: Order-3 Cliffords
+                from_xz(x_to=pY, z_to=pX),  # I-i(+X+Y+Z)
+                from_xz(x_to=mZ, z_to=mY),  # I-i(+X+Y-Z)
+                from_xz(x_to=pZ, z_to=mY),  # I-i(+X-Y+Z)
+                from_xz(x_to=mY, z_to=mX),  # I-i(+X-Y-Z)
+                from_xz(x_to=mZ, z_to=pY),  # I-i(-X+Y+Z)
+                from_xz(x_to=mY, z_to=pX),  # I-i(-X+Y-Z)
+                from_xz(x_to=pY, z_to=mX),  # I-i(-X-Y+Z)
+                from_xz(x_to=pZ, z_to=pY),  # I-i(-X-Y-Z)
+            )
+        return cls._all_single_qubit_cliffords
+
     @property
     def I(cls) -> 'cirq.SingleQubitCliffordGate':
-        if not hasattr(cls, '_I'):
-            cls._I = SingleQubitCliffordGate.from_clifford_tableau(_gate_tableau(1, identity.I))
-        return cls._I
+        return cls.all_single_qubit_cliffords[0]
 
     @property
     def X(cls) -> 'cirq.SingleQubitCliffordGate':
-        if not hasattr(cls, '_X'):
-            cls._X = SingleQubitCliffordGate.from_clifford_tableau(_gate_tableau(1, pauli_gates.X))
-        return cls._X
+        return cls.all_single_qubit_cliffords[1]
 
     @property
     def Y(cls) -> 'cirq.SingleQubitCliffordGate':
-        if not hasattr(cls, '_Y'):
-            cls._Y = SingleQubitCliffordGate.from_clifford_tableau(_gate_tableau(1, pauli_gates.Y))
-        return cls._Y
+        return cls.all_single_qubit_cliffords[2]
 
     @property
     def Z(cls) -> 'cirq.SingleQubitCliffordGate':
-        if not hasattr(cls, '_Z'):
-            cls._Z = SingleQubitCliffordGate.from_clifford_tableau(_gate_tableau(1, pauli_gates.Z))
-        return cls._Z
+        return cls.all_single_qubit_cliffords[3]
 
     @property
     def H(cls) -> 'cirq.SingleQubitCliffordGate':
-        if not hasattr(cls, '_H'):
-            cls._H = SingleQubitCliffordGate.from_clifford_tableau(_gate_tableau(1, common_gates.H))
-        return cls._H
+        return cls.all_single_qubit_cliffords[10]
 
     @property
     def S(cls) -> 'cirq.SingleQubitCliffordGate':
-        if not hasattr(cls, '_S'):
-            cls._S = SingleQubitCliffordGate.from_clifford_tableau(_gate_tableau(1, common_gates.S))
-        return cls._S
+        return cls.all_single_qubit_cliffords[6]
 
     @property
     def CNOT(cls) -> 'cirq.CliffordGate':
         if not hasattr(cls, '_CNOT'):
-            cls._CNOT = CliffordGate.from_clifford_tableau(_gate_tableau(2, common_gates.CNOT))
+            t = qis.CliffordTableau(num_qubits=2)
+            t.xs = [[1, 1], [0, 1], [0, 0], [0, 0]]
+            t.zs = [[0, 0], [0, 0], [1, 0], [1, 1]]
+            cls._CNOT = cls.from_clifford_tableau(t)
```

Comment:
For two-qubit cliffords, I'd suggest explicitly calling the method on `CliffordGate`, since this will be mixed in to the single-qubit class as well:
```suggestion
            cls._CNOT = CliffordGate.from_clifford_tableau(t)
```
Same for CZ and SWAP below.

Reply:
Thanks for catching this! Done.